### ISSUE TITLE
Focal point not very consistent and robust

### ISSUE THREAD

**Issue by PrinceOfAbyss** [timmywil/panzoom#564]
I'm struggling to make the focal point work the way it (at least I think) is supposed to work. Maybe I've taken the whole feature wrong regarding its functionality, but I've studied the source code and I'm almost certain that this is your original intended behavior also, though, in my case, it doesn't work like that...

So, let me be more specific: I'm using your fantastic project to zoom into particular areas of newspaper frontpages, so that the reader, especially the mobile user, will be able to read them easier. I've created this little, intuitive interface that you've seen in another topic I've created, where the user chooses the desired zoom level.

The problem is that on every zoom level change, the zoomed image moves to a default position (_transform-origin: 50% 50% 0px;_). Also, even though I'm making sure to setup correctly the focal point in my code, zooming doesn't honor the code to reflect that in the appropriate CSS rule when applying the transformation; that is something like _transform: scale(2.6) translate(-69.6154px, -42.6923px);_ but rather on every zooming it becomes simply like _transform: scale(2.6) translate(0px);_ making the image "jump" from the previously chosen location to the one set by _transform-origin: 50% 50% 0px;_

I've recorded a short video for you to watch the behavior in action: https://www.youtube.com/watch?v=MgM8LxzKkqE

Also, a working fiddle where you see the code that produces this result can be found here: https://jsfiddle.net/princeofabyss/4cvfau53/

Note in the above fiddle that I'm "reading" the CSS translate x and y values, to use them further while zooming in, but it doesn't seem to work.

Please if that's a miscalculation in your project, help me fix it. Otherwise, if it's something I'm doing wrong in my fiddle, help me spot the mistake... I'm only this close to make this work exactly as I would like it to. Thanks in advance.

**Comment by timmywil** [comment]
Thanks for opening an issue. I think there's a misunderstanding here about what `focal` does. `focal` is a low-level option for focusing in on a point of the element, not relative to the parent dimensions. But zooming the way we usually think about it is often relative to parent dimensions and not the element, which is why Panzoom has the [`zoomToPoint` method](https://github.com/timmywil/panzoom#zoomtopoint) to make things easier. In your fiddle, you're setting the focal to the existing transform, but those numbers aren't really related.

I can't spend too much time working on your fiddle as I'm busy at work, but I don't think this is a bug in Panzoom. The transform-origin is set to the default for modern browsers. That does dictate how an image zooms, but that is default browser behavior that Panzoom works with rather than overrides. The calculation looks right to me or focal point zooming wouldn't work in the demo. But `focal` and `relative` are probably not the options you're looking for.

Also, this fiddle looked very familiar to me. :) I'm sorry I never answered your follow-up question in your previous issue, but the previous fiddle I made doesn't seem to have this issue. I noticed a couple minor issues in that and made some updates. https://jsfiddle.net/timmywil/4rwu6xav/2/ Those include actually saving the Panzoom instance and resetting the image when zoom is set to "OFF".

Finally, your follow-up seemed to be about native scrolling happening while dragging. `preventDefault` on `touchmove` is perfectly fine. Panzoom tries not to get in the way of defaults too much, but it is interesting you had to do that as I don't see that behavior on my phone. Panzoom does set touch-action to none so native scrolling doesn't happen, but perhaps some phones/mobile browsers show and hide the address bar based on other factors and you narrowed it down to `touchmove`.

I hope this clears things up.

**Comment by PrinceOfAbyss** [comment]
This weird behavior that kept resetting the focal point was there even in your first fiddle (https://jsfiddle.net/timmywil/50pve89k/1/)

It’s gone in this fiddle (https://jsfiddle.net/timmywil/4rwu6xav/2/) where you declare const panzoom differently.

```
const panzoom = slide.panzoom || Panzoom(slide, {
    contain: 'outside',
    disableZoom: true
})
```
vs
```
const panzoom = slide.panzoom || (slide.panzoom = Panzoom(slide, {
    contain: 'outside',
    disableZoom: true
}))
```

Pay attention to the following short videos I recorded regarding panzoom's behavior between your two fiddles:

First fiddle: https://www.youtube.com/watch?v=q7FVcPLc3lk
Second fiddle: https://www.youtube.com/watch?v=pJJP_O2jy70

Indeed this was the problem in the first place, and that’s why I had thought that focal point resetting is a stock behavior; that’s why I started messing with focal point settings, in an attempt to try “retrofitting” the functionality myself... Obviously, this was not necessary.

About the preventDefault, I had to put in my code to prevent image panning from moving the whole page up and down; I tried it in a handful of devices, and indeed in some devices it appeared, and in some others it didn’t. So I had to add the code to force the behavior I needed to all of them.

All in all, with this second fiddle you offered, I got a perfectly working zoom plugin. I can’t thank you enough for this. You’re such a kind person, and I hope this kindness you offer so generously, gets back to you one way or the other. Thanks a lot!
